indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -0,0 +1,2828 @@
+        if (checkpoints == null) {
+          try {
+            // catch the exception in failed futures
+            futures.get(i).get();
+          }
+          catch (Exception e) {
+            log.error(e, "Problem while getting checkpoints for task [%s], killing the task", taskId);
+            killTask(taskId, "Exception[%s] while getting checkpoints", e.getClass());
+            taskGroup.tasks.remove(taskId);
+          }
+        } else if (checkpoints.isEmpty()) {
+          log.warn("Ignoring task [%s], as probably it is not started running yet", taskId);
+        } else {
+          taskSequences.add(new Pair<>(taskId, checkpoints));
+        }
+      }
+    }
+    catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    final DataSourceMetadata rawDataSourceMetadata = indexerMetadataStorageCoordinator.getDataSourceMetadata(dataSource);
+
+    if (rawDataSourceMetadata != null && !checkSourceMetadataMatch(rawDataSourceMetadata)) {
+      throw new IAE(
+          "Datasource metadata instance does not match required, found instance of [%s]",
+          rawDataSourceMetadata.getClass()
+      );
+    }
+
+    @SuppressWarnings("unchecked")
+    final SeekableStreamDataSourceMetadata<PartitionIdType, SequenceOffsetType> latestDataSourceMetadata = (SeekableStreamDataSourceMetadata<PartitionIdType, SequenceOffsetType>) rawDataSourceMetadata;
+
+    final boolean hasValidOffsetsFromDb = latestDataSourceMetadata != null &&
+                                          latestDataSourceMetadata.getSeekableStreamPartitions() != null &&
+                                          ioConfig.getStream().equals(
+                                              latestDataSourceMetadata.getSeekableStreamPartitions().getStream()
+                                          );
+    final Map<PartitionIdType, SequenceOffsetType> latestOffsetsFromDb;
+    if (hasValidOffsetsFromDb) {
+      latestOffsetsFromDb = latestDataSourceMetadata.getSeekableStreamPartitions().getPartitionSequenceNumberMap();
+    } else {
+      latestOffsetsFromDb = null;
+    }
+
+    // order tasks of this taskGroup by the latest sequenceId
+    taskSequences.sort((o1, o2) -> o2.rhs.firstKey().compareTo(o1.rhs.firstKey()));
+
+    final Set<String> tasksToKill = new HashSet<>();
+    final AtomicInteger earliestConsistentSequenceId = new AtomicInteger(-1);
+    int taskIndex = 0;
+
+    while (taskIndex < taskSequences.size()) {
+      TreeMap<Integer, Map<PartitionIdType, SequenceOffsetType>> taskCheckpoints = taskSequences.get(taskIndex).rhs;
+      String taskId = taskSequences.get(taskIndex).lhs;
+      if (earliestConsistentSequenceId.get() == -1) {
+        // find the first replica task with earliest sequenceId consistent with datasource metadata in the metadata
+        // store
+        if (taskCheckpoints.entrySet().stream().anyMatch(
+            sequenceCheckpoint -> sequenceCheckpoint.getValue().entrySet().stream().allMat
```

Comment:
Can you remove this commented out code? It's fine for this not to be supported.. it was an edge-case optimization and doesn't affect correctness.